components/scanners/snyk-sbom/internal/transformer/transformer.go

```diff
@@ -0,0 +1,209 @@
+package transformer
+
+import (
+	"context"
+	"crypto/tls"
+	"log"
+	"log/slog"
+	"net/http"
+	"os"
+
+	dtrack "github.com/DependencyTrack/client-go"
+	"github.com/go-errors/errors"
+	"github.com/google/uuid"
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+)
+
+type (
+	// SnykTransformerOption allows customising the transformer.
+	SnykTransformerOption func(g *snykTransformer) error
+
+	snykTransformer struct {
+		rawOutFilePath string
+		apiKey         string
+		apiURL         string
+		projectName    string
+		projectUUID    string
+		projectVersion string
+		debug          bool
+		dtClient       dtrack.Client
+	}
+)
+
+// SnykTransformerWithDTClient allows customising the underlying dependency track client.
+func SnykTransformerWithDTClient(client *dtrack.Client) SnykTransformerOption {
+	return func(g *snykTransformer) error {
+		if client == nil {
+			return errors.Errorf("invalid nil client")
+		}
+		g.dtClient = *client
+		return nil
+	}
+}
+
+// New returns a new snyk transformer.
+func New(opts ...SnykTransformerOption) (*snykTransformer, error) {
+	rawOutFilePath, err := env.GetOrDefault(
+		"RAW_OUT_FILE_PATH",
+		"",
+		env.WithDefaultOnError(false),
```

Comment:
If `env.WithDefaultOnError(false)` is `false` you can omit it as it's the default